### DISCUSSION TITLE
Fill enclosed area

### DISCUSSION THREAD

**Discussion by vakker** [xournalpp/xournalpp#3443] [General] [answered]
Hi,
Is there a tool to fill enclosed area? 
I know I can draw a shape that gets filled if I click on `Fill`, but there's no tool like the paint bucket in MS Paint, right?

**Comment by rolandlo** [comment, marked as answer]
No, there is no such tool.

**Reply by vakker** [reply]
Would that be possible? I can open a feature request if you think it could be useful (it would definitely be useful for me).

**Reply by Technius** [reply]
Xournal++ is a vector graphics application, so the paint bucket tool doesn't fit in well. We can't just color in some region of pixels; instead, we have to create a stroke that is filled with a color. I suppose such a tool could check if the point of activation lies within some closed region and then generate polygon(s) with the appropriate color, but there are a lot of edge cases that would make this tricky to implement.

**Reply by vakker** [reply]
Thanks for the clarification, that makes sense.
